test/api_uber_features.js

```diff
@@ -86,6 +91,62 @@ test('api no change null', function(t) {
     });
 });
 
+fs.writeFileSync(fileFour, 'require("./less.less")');
+fs.writeFileSync(lessOne, '@import "./less2.less";');
+fs.writeFileSync(lessTwo, 'html { color: red; }');
+
+test('recompile transform dependents', function(t) {
+    t.plan(9);
+
+    var cacheFile = path.join(tmpdir, 'transform.cache.json');
+    var w = watchify(browserify(fileFour, watchify.args()), {
+        cacheFile: cacheFile,
+        watch: false
+    })
+        .transform(lessify);
+
+    w.bundle(function (err, src) {
+        t.ifError(err);
+        t.notEqual(src.toString('utf8').indexOf('color:red'), -1);
+
+        w.close();
+        w.write();
+
+        setTimeout(function () {
+            fs.writeFile(lessTwo, 'html { color: green; }', function (err) {
+                t.ifError(err);
+
+                w = watchify(browserify(fileFour, watchify.args()), {
+                    cacheFile: cacheFile,
+                    watch: false
+                })
+                    .transform(lessify);
+
+                w.bundle(function (err, src) {
+                    t.ifError(err);
+                    t.assert(src, 'rebundled');
+                    t.notEqual(src.toString('utf8').indexOf('color:green'), -1);
```

Comment:
Yes, this happens because it actually does have the `cache` object. So it rebuilds everything from scratch. On line `119` you actually need to give `browserify` a `cache` argument that contains the actual cache object. You could do `browserify(fileFour, xtend(watchify.args(), {cache: cacheObj}))`. 
`browserify` takes the `cache` to use internally, and `watchify` takes the `cacheFile` file path to know where to write this internal object to afterwards. 

If you can make that change, I'll merge and release.